src/Messaging/MessageBuffer.php

```diff
@@ -157,7 +157,7 @@ public function frameCheck(FrameInterface $frame) {
                         return $this->newCloseFrame(Frame::CLOSE_BAD_PAYLOAD);
                     }
 
-                    return $this->newCloseFrame(Frame::CLOSE_NORMAL);
+                    return $this->newCloseFrame($closeCode);
```

Comment:
I believe you commented on an outdated diff. We're not passing back the original close frame passed in to the check. I don't see any harm as the payload is not externally mutable. This maintains the full frame passed (code and reason).